blog/src/a-base32-checksum.md

```diff
@@ -0,0 +1,1241 @@
+bits per char).
+
+While they can be extended to larger alphabets, that is not trivial nor
+standard. (Ah, the joy of tweaking the Verhoeff parameters.) Worse, they are no
+longer the coolest kids on the block.
+
+I found a 2016 [paper][Chen 2016] referencing a 2014 one (which I couldn't get
+my hands on. Curse the IEEE overseers and their labyrinthic subscription
+models! _Please pay $84.50 membership and the paper is $14.95, oh and would you
+mind donating because IEEE helps improve the human condition? Also good news, we
+offer free shipping for PDF documents._)
+
+No joke, but I digress. The 2014 paper has no value anyway, as the principle is
+described in full in the [2016 paper][Chen 2016], available online.
+
+The main insight is this: if your check digit a<sub>n+1</sub> solves Σ aᵢ·Pⁱ = 0
+on a finite field where P is the companion matrix of a primitive polynomial,
+*it suddenly detects substitutions, transpositions, and twin substitutions* with
+up to R characters between them, where R grows with the cardinal. Magic!
+
+(We will dig more into exactly what errors it detects later.)
+
+You must pick two numbers carefully to design your checksum: the cardinal and
+the primitive.
+
+- The **cardinal** is the number of elements in the finite field. It is
+  therefore also the number of possible checksums. For a single-character check
+  digit, it must be smaller than the size of your alphabet.
+- The **primitive** does not matter all that much. The most significant part
+  about it is that you need to compute one. Unfortunately, brute-force is usually
+  the simplest way to find some.
+
+There are two possible cardinals: *prime numbers* and *prime powers*. Prime
+cardinals are neat, because operations are scalar. On the other hand, additions
+and multiplications on prime power cardinal fields are matricial.
+
+I first wanted a single-character checksum. In order to use all the bits that
+this checksum had at its disposal, I needed it to support the full base32
+alphabet. That meant the cardinal needed to be 32. Sadly, 32 is not a prime; so
+it needed to be a prime power. If 2 is the prime, 2⁵ is the cardinal.
+
+Since the cardinal is not a prime, P is a matrix. Here, its size is 5×5.
+To build it, we can pick any order-5 [primitive polynomial][]; I chose 1+x²+x⁵.
+To get the companion matrix, set a diagonal of 1s, and set the rightmost column
+with the polynomial coefficients, starting with x⁰:
+
+    ⎛ 0 0 0 0 1 ⎞
+    ⎜ 1 0 0 0 0 ⎟
+    ⎜ 0 1 0 0 1 ⎟
+    ⎜ 0 0 1 0 0 ⎟
+    ⎝ 0 0 0 1 0 ⎠
+
+Each aᵢ term is a representation of a base32 character as a GF(2⁵) (finite field
+of order 2⁵) polynome: a is 0, which we represent as (0 0 0 0 0); b is 1 or (0 0
+0 0 1); f is 1+x² or (0 0 1 0 1).
+
+Validating a check character is straightforward: compute Σ aᵢ·Pⁱ; verify that
+the result is a zero vector. It does involve writing custom matrix
+multiplication and addition primitives to ensure that they are all computed
```

Comment:
something sounds weird to me here, looks like `addition` is used as a verb - did you mean `writing custom matrices, multiplying and summing up primitives to ensure...`? 
(I'm totally out of depth technically for this section of the article to make sense to me so please be kind if my comment appears overly naive or off-topic 💃 )

Reply:
The way it is meant to be parsed is `writing (custom matrix (multiplication and addition)) primitives`. It uses the programmatic meaning of "primitive" (a simple operation relied upon to compute larger algorithms), although it is confusing here, since we use the mathematical meaning (which is very different) in a previous paragraph.

Reply:
Got it, makes sense now that you lay it out for me - maybe readers who are more experienced in these types of technical posts will have no problem parsing this (but the simpler version works better for me) 👍